pymatgen/io/vasp/outputs.py

```diff
@@ -658,21 +673,25 @@ def run_type(self):
             rt = "B3LYP"
         elif self.parameters.get("LHFCALC", True):
             rt = "PBEO or other Hybrid Functional"
-        elif self.parameters.get("LUSE_VDW", False):
-            if self.incar.get("METAGGA", "").strip().upper() in METAGGA_TYPES:
-                rt = METAGGA_TYPES[self.incar.get("METAGGA", "").strip().upper()] + "+rVV10"
-            else:
-                rt = GGA_TYPES[self.parameters.get("GGA", "").strip().upper()] + "+rVV10"
-        elif self.incar.get("METAGGA", "").strip().upper() in METAGGA_TYPES:
-            rt = METAGGA_TYPES[self.incar.get("METAGGA", "").strip().upper()]
-            if self.is_hubbard or self.parameters.get("LDAU", True):
-                rt += "+U"
+        elif self.incar.get("METAGGA") and self.incar.get("METAGGA") != "--":
+            incar_tag = self.incar.get("METAGGA", "").strip().upper()
+            rt = METAGGA_TYPES.get(incar_tag, incar_tag)
+        elif self.parameters.get("GGA"):
+            incar_tag = self.parameters.get("GGA", "").strip().upper()
+            rt = GGA_TYPES.get(incar_tag, incar_tag)
         elif self.potcar_symbols[0].split()[0] == 'PAW':
             rt = "LDA"
-        elif self.parameters.get("GGA", "").strip().upper() in GGA_TYPES:
-            rt = GGA_TYPES[self.parameters.get("GGA", "").strip().upper()]
-            if self.is_hubbard or self.parameters.get("LDAU", True):
-                rt += "+U"
+
```

Comment:
We should add a

```python
else:
    rt = "unknown"
    warnings.warn("Unknown Run type!")
```

Reply:
@shyuep I added this. Realistically I'm not sure it will ever be triggered, but it's good to catch anyway. Right now if the contents of `METAGGA` or `GGA` are not found in the respective dicts, the code will just pass through whatever that content is to the `run_type`. So if `vasprun.xml` contains `GGA=RANDOMFUNCTIONAL`, the `run_type` that would be returned is `RANDOMFUNCTIONAL`.

So, the only way the `unknown` would ever be triggered were if the `GGA` tag were missing from `vasprun.xml`. I'm not sure it's possible for that to happen and still have a valid .xml file, but again, it's good to cover all bases.

Reply:
If the GGA tag is missing it defaults to whatever functional is defined in the POTCAR. I'm not sure how that would change the logic here.

Thanks for adding the extra lines @rkingsbury! Appreciate the clean-up.

Reply:
I think `vasprun.xml` contains GGA and METAGGA tags even if they are not set in the INCAR. For example, I know that GGA calculations still have `METAGGA=--` in the file. As you say, if GGA is not in the INCAR VASP will default to something, but I think it will always put that something in the vasprun.xml file. Could be wrong though.

Reply:
I think the catch is in case VASP implements new functionals right?

Reply:
I think Ryan's point is the the `GGA_TYPES` variable is just to map the VASP GGA tag names to a human-readable name (e.g. PS -> PBEsol). If it's not in GGA_TYPES because VASP implements a new functional, you would just get the raw VASP tag name instead.